### PR TITLE
#4252 ensure surefire always triggers in test phase for all modules

### LINKED ISSUE #4252: Maven builds not running all tests
PR #4248 failed in the Gradle build in https://github.com/odpi/egeria/pull/4248/checks?check_run_id=1485869158
```
> Task :open-metadata-implementation:adapters:open-connectors:integration-connectors:files-integration-connectors:test FAILED

Gradle suite > Gradle test > org.odpi.openmetadata.adapters.connectors.integration.basicfiles.ffdc.AuditCodeTest > testAllAuditCodeValues FAILED
    java.lang.AssertionError at AuditCodeTest.java:24


FAILURE: Build failed with an exception.
```

Yet the maven build completes successfully.

On closer examination we see in https://github.com/odpi/egeria/pull/4248/checks?check_run_id=1485869159:
```
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ files-integration-connectors ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
```

So it would seem the gradle build ran the tests, maven did not.

Need to investigate to see if this is just this component, or if we are missing other tests too.

cc: @mandy-chessell

### PR DESCRIPTION
**Pull request by planetf1**
Signed-off-by: Nigel Jones <nigel.l.jones+git@gmail.com>

Fixes #4252 

 - Adds surefire as a default build plugin for all modules
 - surefire is intelligent enough to skip pom modules & others without tests

### CODE REVIEW

**Comment by planetf1** [comment]
Fixed some of the duplicationsin repository services ... more to follow (which is why test currently still failing)

**Comment by planetf1** [comment]
@mandy-chessell Errors are unique (there were A LOT). I'm not convinced my wording is great, so any review welcome.
However, they are now all unique and pass the unit test correctly, which is invoked via maven (I will look at gradle as part of the gradle thread - need to check cross-tool consistency)

**Comment by planetf1** [comment]
The repository services UTs are fixed (de-duplicated). However this PR now fails with:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project ui-chassis-spring: There are test failures.
```

**Comment by planetf1** [comment]
@bogdan @sarbull - Note the disabling of the spring ui chassis tests. In fact they were never running  it would seem (just like Mandy's test). I opened up the referenced issue to track fixing them (and reenabling them) and assigned to you. Hope that works.

**Comment by planetf1** [comment]
Note #4294 adds fixes for the UI UT's . I would be desirable to get this change in first. I suggest merging this tomorrow (and raise a new PR if further changes needed to message texts)

File: pom.xml [hunk 1]
```diff
@@ -3156,6 +3156,13 @@
                 </configuration>
             </plugin>
 
+            <!-- Use surefire for unit tests -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.1</version>
```
Comment (bogdan-sava) [inline pom.xml:3163]: overriding version from main pom.xml ?

**Review by bogdan-sava — CHANGES_REQUESTED** [review]

File: open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/pom.xml [hunk 2]
```diff
@@ -334,6 +322,14 @@
             <groupId>org.springframework.cloud</groupId>
             <artifactId>spring-cloud-netflix-zuul</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.1</version>
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-implementation/user-interfaces/ui-chassis/ui-chassis-spring/pom.xml:329]: Thanks - removed.. though there is another issue open to correct the ui tests. But in any case this is against our strategy of keeping versions in dependencyManagement so thanks for the spot!

File: pom.xml (hunk 1 above)
Comment (planetf1) [reply pom.xml]: Good spot. Bad cut/paste on my behalf. Thanks. Fixed

**Comment by planetf1** [comment]
@bogdan I found a few others so updated there also -- we still have a couple
 - for gaian/ranger plugin - these are moving to new repo anyway
 - specific dependencies of plugins where dependencyManagement didn't seem to work (like groovy for fvt)

These I'm sure can be improved in future

but I removed some additional obvious ones ie for jupiter api which were unnecessary.

**Comment by planetf1** [comment]
@bogdan-sava I think I've answered your questions. I'm planning to merge this shortly so that we can improve the testing integrity

**Pull request merged**